src/observers/stores/argilla.py

```diff
@@ -112,3 +113,28 @@ def add(self, record: "Record") -> None:
 
         record_dict = {k: v for k, v in record_dict.items() if k in self._dataset_keys}
         self._dataset.records.log([record_dict])
+
+    async def add_async(self, record: "Record"):
+        """
+        Add a new record to the database asynchronously
+
+        Args:
+            record (`Record`):
+                The record to add to the database.
+        """
+        if not self._dataset:
+            self._init_table(record)
+
+        record_dict = asdict(record)
+
+        for text_field in record.text_fields:
+            if text_field in record_dict:
+                record_dict[f"{text_field}_length"] = len(record_dict[text_field])
+
+        record_dict = {k: v for k, v in record_dict.items() if k in self._dataset_keys}
+        # Use argilla's native async API
+        await self._dataset.records.alog(
+            [record_dict],
+            background=True,
+            verbose=False,
+        )
```

Comment:
I don't think `alog` exists, but let me look at the Argilla integration in a separate PR. Perhaps we can now default to the threading method as implemented in the other stores?